dspace-spring-rest/src/test/java/org/dspace/app/rest/RelationshipTypeRestControllerIT.java

```diff
@@ -0,0 +1,212 @@
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.dspace.app.rest.builder.CollectionBuilder;
+import org.dspace.app.rest.builder.CommunityBuilder;
+import org.dspace.app.rest.builder.ItemBuilder;
+import org.dspace.app.rest.builder.RelationshipBuilder;
+import org.dspace.app.rest.matcher.EntityTypeMatcher;
+import org.dspace.app.rest.matcher.PageMatcher;
+import org.dspace.app.rest.matcher.RelationshipMatcher;
+import org.dspace.app.rest.matcher.RelationshipTypeMatcher;
+import org.dspace.app.rest.test.AbstractEntityIntegrationTest;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.EntityType;
+import org.dspace.content.Item;
+import org.dspace.content.Relationship;
+import org.dspace.content.RelationshipType;
+import org.dspace.content.service.EntityTypeService;
+import org.dspace.content.service.RelationshipTypeService;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class RelationshipTypeRestControllerIT extends AbstractEntityIntegrationTest {
+
+    @Autowired
+    private RelationshipTypeService relationshipTypeService;
+
+    @Autowired
+    private EntityTypeService entityTypeService;
+
+    @Test
+    public void findAllEntityTypes() throws Exception {
+
+        context.turnOffAuthorisationSystem();
+
+        getClient().perform(get("/api/core/entitytypes"))
+
+                   .andExpect(status().isOk())
+                   .andExpect(jsonPath("$.page",
+                                       is(PageMatcher.pageEntryWithTotalPagesAndElements(0, 20, 1, 7))))
+                   .andExpect(jsonPath("$._embedded.entitytypes", containsInAnyOrder(
+                       EntityTypeMatcher.matchEntityTypeEntryForLabel("Publication"),
+                       EntityTypeMatcher.matchEntityTypeEntryForLabel("Person"),
+                       EntityTypeMatcher.matchEntityTypeEntryForLabel("Project"),
+                       EntityTypeMatcher.matchEntityTypeEntryForLabel("OrgUnit"),
+                       EntityTypeMatcher.matchEntityTypeEntryForLabel("Journal"),
+                       EntityTypeMatcher.matchEntityTypeEntryForLabel("JournalVolume"),
+                       EntityTypeMatcher.matchEntityTypeEntryForLabel("JournalIssue")
+
+                   )))
+        ;
+    }
+
+    @Test
+    public void findAllRelationshipTypesForPublications() throws Exception {
+
+        context.turnOffAuthorisationSystem();
```

Comment:
this is not necessary (and it was not restored)

Reply:
This has been resolved in https://github.com/atmire/DSpace/commit/4c6e939e57352acc854053302df37f78c995571a